ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/encoding/TransactionDecoder.java

```diff
@@ -109,6 +109,9 @@ private static Transaction decodeTypedTransaction(
   public static Transaction decodeOpaqueBytes(
       final Bytes opaqueBytes, final EncodingContext context) {
     var transactionType = getTransactionType(opaqueBytes);
+    if (opaqueBytes.isEmpty()) {
```

Comment:
Do you think we should do the test before `getTransactionType`? 
Also, I would suggest changing the exception message to "Empty opaque bytes"